button-merchant/src/main/java/com/usebutton/merchant/ButtonInternalImpl.java

```diff
@@ -68,12 +72,20 @@
      */
     private final AtomicBoolean hasReceivedDirectDeeplink = new AtomicBoolean();
 
+    private static final Pattern APP_ID_PATTERN = Pattern.compile("^app-[0-9a-zA-Z]+$");
+
     ButtonInternalImpl(Executor executor) {
         this.attributionTokenListeners = new ArrayList<>();
         this.executor = executor;
     }
 
     public void configure(ButtonRepository buttonRepository, String applicationId) {
+        if (!APP_ID_PATTERN.matcher(applicationId).matches()) {
+            Log.e(TAG, "Application ID [" + applicationId + "] is not valid. "
+                    + "You can find your Application ID in the dashboard by logging in at"
+                    + " https://app.usebutton.com/");
+        }
+
         buttonRepository.setApplicationId(applicationId);
```

Comment:
If the app Id is not hostname compatible, the library is rendered useless

Reply:
Good point. I'll update it to prevent configuration if invalid App ID is passed.